transaction - reply with read_only if an ongoing transaction gets an external commit/prepare::Request

albinsjolund opened the pull request:
Before:
Transaction manager sometimes tried to commit stuff that had already been handled
when an external resource sent a request.

lazan commented on middleware/transaction/unittest/source/test_manager.cpp:
```diff
@@ -1634,6 +1634,262 @@ namespace casual
+            message.process = rm2.process;
+            message.trid = branch;
+            local::send::tm( message);
+         }
```
This is not just "gateway involved"

lazan commented on middleware/transaction/unittest/source/test_manager.cpp:
```diff
@@ -1634,6 +1634,262 @@ namespace casual
+            message.process = process::handle();
+
+            local::send::tm( message);
+         }
```
What does "commit::Request" mean? I think you need to describe that you act as "user"/tx_begin()

lazan commented on middleware/transaction/unittest/source/test_manager.cpp:
```diff
@@ -1634,6 +1634,262 @@ namespace casual
+            reply.trid = message.trid;
+
+            local::send::tm( reply);
+         }
```
... same here.

lazan commented on middleware/transaction/unittest/source/test_manager.cpp:
```diff
@@ -1634,6 +1634,262 @@ namespace casual
+            reply.trid = message.trid;
+
+            local::send::tm( reply);
+         }
```
I think you need to explain what this is, like "act like another domains TM that got the resource::prepare::Request ....." and so on.
The comment now doesn't really say anything.

lazan replied:
Perfekt!

lazan reviewed (approved):
I approve this, but I really think you need to document/comment the unittests better from a logical perspective. Like: "we act as resource 1, and receive ...bla bla " "we act as outbound (resource) and then send ... bla bla"

albinsjolund committed:
transaction - reply with read_only if an ongoing transaction gets an external commit/prepare::Request

Before:
Transaction manager sometimes tried to commit stuff that had already been handled
when an external resource sent a request.

lazan reviewed (approved):
Much better!
